pennylane/_qubit_device.py

```diff
@@ -401,6 +430,129 @@ def execute_new(self, circuit, **kwargs):
         #     self.tracker.record()
         return results
 
+    def shot_vec_statistics(self, circuit):
+        """Computes statistics for a shot vector.
+
+        This method is used by execute_new.
+
+        Args:
+            circuit (~.tapes.QuantumTape): circuit to execute on the device
+
+        Raises:
+            QuantumFunctionError: if the value of :attr:`~.Observable.return_type` is not supported
+
+        Returns:
+            tuple: stastics for each shot item from the shot vector
+        """
+        results = []
+        s1 = 0
+
+        ret_types = [m.return_type for m in circuit.measurements]
+        counts_exist = any(ret is qml.measurements.Counts for ret in ret_types)
+        single_measurement = len(circuit.measurements) == 1
+
+        for shot_tuple in self._shot_vector:
+            s2 = s1 + np.prod(shot_tuple)
+            r = self.statistics_new(
```

Comment:
I think we are missing some calls `self._asarray(res, dtype=self.R_DTYPE)` for measurements that are not counts or sample. But as suggested we should make those changes in the `statistics_new` function

Reply:
I've moved them there, is it looking good now?